mesonpy/__init__.py

```diff
@@ -944,18 +947,21 @@ def _validate_string_collection(key: str) -> None:
                 ))
             )
 
-    _validate_string_collection('dist-args')
-    _validate_string_collection('setup-args')
-    _validate_string_collection('compile-args')
-    _validate_string_collection('install-args')
+    meson_args_keys = typing_get_args(MesonArgsKeys)
+    meson_args_cli_keys = tuple(f'{key}-args' for key in meson_args_keys)
+
+    for key in config_settings:
+        if key not in ('builddir', *meson_args_cli_keys):
```

Comment:
It does not matter much, but this could be a set instead than a tuple.

Reply:
Yeah, sure, I tend to use tuples in these situations as they are less expensive (not like it would have any effect in this case in specific anyway :rofl:, it's just my general rule) and we are not handling any user data, so if there's duplication, it's incorrect code.